src/coll/error.rs

```diff
@@ -114,55 +104,44 @@ impl fmt::Display for BulkWriteException {
 
 impl fmt::Display for BulkWriteError {
     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        try!(write!(
+        write!(
             fmt,
             "BulkWriteError at index {} (code {}): {}\n",
             self.index,
             self.code,
             self.message
-        ));
+        )?;
 
         match self.request {
-            Some(ref request) => try!(write!(fmt, "Failed to execute request {:?}\n.", request)),
-            None => {
-                try!(write!(
-                    fmt,
-                    "No additional error information was received.\n"
-                ))
-            }
+            Some(ref request) => write!(fmt, "Failed to execute request {:?}\n.", request),
+            None => fmt.write_str("No additional error information was received.\n")
         }
-
-        Ok(())
     }
 }
 
 impl WriteException {
     /// Returns a new WriteException containing the given errors.
     pub fn new(wc_err: Option<WriteConcernError>, w_err: Option<WriteError>) -> WriteException {
-        let mut s = match wc_err {
-            Some(ref error) => format!("{:?}\n", error),
-            None => String::from(""),
-        };
+        use std::fmt::Write;
+
+        let mut s = wc_err.as_ref().map(|error| format!("{:?}\n", error)).unwrap_or_default();
 
         if let Some(ref error) = w_err {
-            s.push_str(&format!("{:?}\n", error)[..]);
+            write!(s, "{:?}\n", error).expect("can't format error");
         }
```

Comment:
I don't think those are equivalent, actually, would it compile? `w_err.as_ref().map(ToString::to_string).unwrap_or_default()` would, but it would:
 – result in one more allocation, and
 – format using `Display` instead of `Debug`.